src/scope/core/pipelines/interface.py

```diff
@@ -13,7 +17,65 @@ class Requirements(BaseModel):
 
 
 class Pipeline(ABC):
-    """Abstract base class for all pipelines."""
+    """Abstract base class for all pipelines.
+
+    Pipelines must implement get_config_class() to return their Pydantic config model.
+    This enables:
+    - Validation via model_validate() / model_validate_json()
+    - JSON Schema generation via model_json_schema()
+    - Type-safe configuration access
+    - API introspection and automatic UI generation
+
+    See schema.py for the BasePipelineConfig model and pipeline-specific configs.
+    For multi-mode pipeline support (text/video), use multi_mode.MultiModePipeline
+    as the base class which provides declarative mode configuration.
+    """
+
+    @classmethod
+    def get_config_class(cls) -> type["BasePipelineConfig"]:
+        """Return the Pydantic config class for this pipeline.
+
+        The config class should inherit from BasePipelineConfig and define:
+        - pipeline_id: Unique identifier
+        - pipeline_name: Human-readable name
+        - pipeline_description: Capabilities description
+        - pipeline_version: Version string
+        - Default parameter values for the pipeline
+
+        Returns:
+            Pydantic config model class
+
+        Note:
+            Subclasses should override this method to return their config class.
+            The default implementation returns BasePipelineConfig.
+
+        Example:
+            from .schema import LongLiveConfig
+
+            @classmethod
+            def get_config_class(cls) -> type[BasePipelineConfig]:
+                return LongLiveConfig
+        """
+        from .schema import BasePipelineConfig
+
+        return BasePipelineConfig
+
+    @classmethod
+    def get_schema(cls) -> dict[str, Any]:
```

Comment:
The caller can just get this information from the value returned by `get_config_class()` so seems that we can simplify and not introduce an additional required method here.